woocommerce/woocommerce
# Show correct decimals on product price and show the sale price of products

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes 18825-gh-Automattic/woocommerce.com and 21785-gh-Automattic/woocommerce.com.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure your WCCOM environment is running on this [PR](21705-gh-Automattic/woocommerce.com) branch 
2. Change product prices to have some decimal values for couple of products  on your WCCOM local and make couple of products on sale
3. clear cache of your WCCOM local `npm run wp:cli cache flush`
4. Next you'll need to point your WC Core environment to your local WCCOM by following instruction on this PR https://github.com/woocommerce/woocommerce/pull/51170
5. Go to  [in-app marketplace](http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fextensions).
6. view extensions and Discover tab
7. confirm the products price with decimal values are displayed correctly
8. Also confirm the sale price is shown similar to what we have on WCCOM
![Screenshot 2024-09-18 at 12 58 08 PM](https://github.com/user-attachments/assets/27d2794e-fde8-47cc-94b4-d44e1c8b0a46)
![Screenshot 2024-09-18 at 12 58 02 PM](https://github.com/user-attachments/assets/3cb70976-c73c-4301-bd0a-4cead14afbe7)

9. Switch VO on mac and confirm the product prices on card are read out correctly
10. Bonus: Test this change doesn't break anything on IAM for older versions



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Show the sale price and display the correct number of decimals for the product card price.
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
